### PR TITLE
Add cache pkg

### PR DESCRIPTION
**Pull request by adrianchiris**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+github.com/fsnotify/fsnotify v1.4.7/go.mod h1:jwhsz4b93w/PPRr/qN1Yymfu8t87LnFCMoQvtojpjFo=
+github.com/golang/protobuf v1.2.0/go.mod h1:6lQm79b+lXiMfvg/cZm0SGofjICqVBUtrP5yJMmIC1U=
+github.com/hpcloud/tail v1.0.0 h1:nfCOvKYfkgYP8hkirhJocXT2+zOD8yUNjXaWfTlyFKI=
+github.com/hpcloud/tail v1.0.0/go.mod h1:ab1qPbhIpdTxEkNHXyeSf5vhxWSCs/tWer42PpOxQnU=
+github.com/onsi/ginkgo v1.6.0/go.mod h1:lLunBs/Ym6LB5Z9jYTR76FiuTmxDTDusOGeTQH+WWjE=
+github.com/onsi/ginkgo v1.12.0 h1:Iw5WCbBcaAAd0fpRb1c9r5YCylv4XDoCSigm1zLevwU=
+github.com/onsi/ginkgo v1.12.0/go.mod h1:oUhWkIvk5aDxtKvDDuw8gItl8pKl42LzjC9KZE0HfGg=
+github.com/onsi/gomega v1.7.1 h1:K0jcRCwNQM3vFGh1ppMtDh/+7ApJrjldlX8fA0jDTLQ=
+github.com/onsi/gomega v1.7.1/go.mod h1:XdKZgCCFLUoM/7CFJVPcG8C1xQ1AJ0vpAezJrB7JYyY=
+github.com/onsi/gomega v1.9.0 h1:R1uwffexN6Pr340GtYRIdZmAiN4J+iw6WG4wog1DUXg=
```
Comment (moshe010) [inline go.sum]: why it 1.9.0 and in above it 1.7.1?

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+var (
+	// Used for caching CNI network state
+	DefaultCacheDir = "/var/lib/cni/rdma"
```
Comment (moshe010) [inline pkg/cache/cache.go]: can it be const?

File: pkg/cache/cache.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	path := filepath.Join(sc.basePath, sRef)
+	bytes, err := sc.fsOps.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("failed to read cache data in the path(%q): %v", path, err)
+	}
+	err = json.Unmarshal(bytes, state)
+	if err != nil {
+		return err
+	}
+	return err
```
Comment (moshe010) [inline pkg/cache/cache.go]: you don't nerr Line 71-74, just return json.Unmarshal(bytes, state)

File: pkg/cache/cache.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/cache/cache.go]: I wanted to keep an option to make it configurable, shall i just rename to just `CacheDir` ?

File: go.sum (hunk 1 above)
Comment (adrianchiris) [reply go.sum]: ginkgo 1.12.0 depends on gomega 1.7.1 see [1] 

[1] https://github.com/onsi/ginkgo/blob/v1.12.0/go.mod

File: pkg/cache/cache.go (hunk 3 above)
Comment (adrianchiris) [reply pkg/cache/cache.go]: DOne

**Pull request merged**